tests/x11/yast2_network.pm

```diff
@@ -0,0 +1,67 @@
+use base "x11test";
+use testapi;
+
+# Helper method to enable routing through `lan` client.
+sub enable_routing() {
+  send_key "alt-u"; # Routing tab
+  assert_screen "yast2_network-routing_disabled";
+  send_key "alt-i"; # Activate IPv4 routing
+  send_key "alt-p"; # Activate IPv6 routing
+  assert_screen "yast2_network-routing_enabled";
+}
+
+# Helper method to check if routing is enabled.
+sub check_routing_enabled_ui() {
+    # Check at UI level
+    send_key "alt-u"; # Routing
+    assert_screen "yast2_network-rounting_enabled";
+}
+
+sub check_routing_enabled_console() {
+    # Check at system level
+    script_run "sysctl net.ipv4.conf.all.forwarding > /dev/$serialdev";
+    wait_serial("net.ipv4.conf.all.forwarding = 1", 5) || die("IPv4 not enabled");
+}
+
+sub start_yast2_lan() {
+    script_run "yast2 lan";
+    assert_screen "yast2_network-lan", 50;
+}
+
+# Test for basic yast2-network functionality.
+sub run() {
+  my $self = shift;
+
+  # Make sure yast2-network ins installed (if not, install it)
+  ensure_installed "yast2-network";
+
+  # Start xterm as root
+  x11_start_program("xterm");
+  wait_idle;
+  become_root;
+
+  # Make sure that firewall is running
+  script_run "yast2 firewall enable";
+
+  # Enable routing
+  $self->start_yast2_lan();
```

Comment:
JFYI (assuming you're not perl native :) if you don't need the object in the function, you can skip the $self. $self->start is just syntactic sugar for start($self)